src/System.Windows.Forms/src/System/Windows/Forms/ListView.cs

```diff
@@ -6926,21 +6926,25 @@ protected override void WndProc(ref Message m)
                 case User32.WM.KEYUP:
                     int key = (int)m.WParamInternal;
 
-                    // User can collapse/expand a group using the keyboard by focusing the group header and using left/right
+                    // User can collapse/expand a group using the keyboard by focusing the group header and using left/right.
                     if (GroupsDisplayed && (key is User32.VK.LEFT or User32.VK.RIGHT) && SelectedItems.Count > 0)
                     {
-                        ListViewGroup group = SelectedItems[0].Group;
-
-                        if (group is null || group.CollapsedState is ListViewGroupCollapsedState.Default)
+                        // User can select more than one group.
+                        HashSet<int> groups = new();
+                        foreach (ListViewItem selectedItem in SelectedItems)
                         {
-                            break;
-                        }
+                            ListViewGroup group = selectedItem.Group;
+                            if (group is null || !groups.Add(group.ID) || group.CollapsedState is ListViewGroupCollapsedState.Default)
```

Comment:
I'd keep the original order and append the new operation to the end as it has perf implications
```suggestion
                            if (group is null || group.CollapsedState is ListViewGroupCollapsedState.Default || !groups.Add(group.ID))
```